### ISSUE TITLE
Warning with Next.js 10.2.0 and SSR

### ISSUE THREAD

**Issue by medihack** [clauderic/dnd-kit#285]
First, thank you for this great library! Unfortunately, I get a new warning after I upgraded to the latest Next.js version (v10.2.0) because of the server-side rendering in development mode: `Warning: Prop 'aria-describedby' did not match. Server: "DndDescribedBy-2" Client: "DndDescribedBy-1"` (full traceback below). It seems that the server-side ID increases each time I refresh the page (maybe because of the new stuff the Next.js team added to improve startup times).

I see that the ID is generated automatically with [`useUniqueId`](https://github.com/clauderic/dnd-kit/blob/f9ec28fed77778669f93cfecee159dba54db38b4/packages/core/src/components/DndContext/DndContext.tsx#L168) and without allowing to set it manually. A possible way to fix this I guess would be an optional `id` prop for the `DndContext` and pass it to `useUniqueId` to set the `draggableDescribedById` manually. What do you think?

Full traceback:
```
Warning: Prop `aria-describedby` did not match. Server: "DndDescribedBy-2" Client: "DndDescribedBy-1"
    at div
    at SortableItem (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:308602:17)
    at SortableContext (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:5647:3)
    at DndContext (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:4498:3)
    at nav
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at List (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:25308:79)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Paper (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:30453:80)
    at SectionList (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:307579:86)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:10205:99)
    at form
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:10205:99)
    at FormProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:125662:147)
    at StructureForm (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:308867:23)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:10205:99)
    at StructurePanel (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:309086:114)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:10205:99)
    at TransformerRegistry (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:309831:23)
    at ModuleRegistry (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:303510:23)
    at StructuredReportContent
    at TemplatePage
    at WithSuperJSON (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:324597:42)
    at BlitzInnerRoot (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:1300:5)
    at main
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:10205:99)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Container (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:15860:80)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:10205:99)
    at Layout (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:311278:20)
    at Suspense
    at ErrorBoundary (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:125435:37)
    at InnerThemeProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:50374:67)
    at ThemeProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:54474:5)
    at ThemeProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:50394:5)
    at SwitchableThemeProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:63646:23)
    at LocalizationProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:54316:5)
    at LocalizationProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:63404:23)
    at App (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:66009:24)
    at AppWithStructuredReportTranslations (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:64385:28)
    at AppWithSiteTranslations (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:64235:28)
    at Provider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:131913:20)
    at AppWithRedux (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:64099:32)
    at Hydrate (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:131181:23)
    at QueryClientProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:131256:21)
    at BlitzProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:1079:21)
    at BlitzOuterRoot (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:1331:34)
    at ErrorBoundary (http://localhost:3000/_next/static/chunks/main.js?ts=1621372663479:142:47)
    at ReactDevOverlay (http://localhost:3000/_next/static/chunks/main.js?ts=1621372663479:245:23)
    at Container (http://localhost:3000/_next/static/chunks/main.js?ts=1621372663479:7839:5)
    at AppContainer (http://localhost:3000/_next/static/chunks/main.js?ts=1621372663479:8328:24)
    at Root (http://localhost:3000/_next/static/chunks/main.js?ts=1621372663479:8464:25)
```

**Comment by DiegoMcDipster** [comment]
I have the same issue.  

To recreate:
1. npx create-next-app dnd-kit-test
2. Take the code from : https://codesandbox.io/s/dnd-kit-flex-wrap-layout-yt4rv
3. npm run dev

**Comment by medihack** [comment]
I could fix this error by creating an optional id prop for `DndContext`. Unfortunately, there is another error message then (stack trace below), because of the way [@dnd-kit uses createPortal in `Accessibility.tsx`](https://github.com/clauderic/dnd-kit/blob/fd7c173da4ca21b4ea3341805100bcd676a4a532/packages/core/src/components/Accessibility/Accessibility.tsx#L46) which is incompatible with how it [should be used in Next.js](https://github.com/vercel/next.js/blob/canary/examples/with-portals/components/ClientOnlyPortal.js). Because of the `canUseDOM` query it produces `null` on the server, but a real element on the client.

```
Warning: Prop `id` did not match. Server: "null" Client: "foo"
    at div
    at HiddenText (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621448429258:24:3)
    at Accessibility (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621448429258:928:3)
    at DndContext (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621448429258:2009:3)
    at nav
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:3284:73
    at List (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:25208:86)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:3284:73
    at Paper (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:30371:86)
    at SectionList (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621448429258:393589:93)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:3284:73
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:10052:106)
    at form
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:3284:73
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:10052:106)
    at FormProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:122242:150)
    at StructureForm (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621448429258:394870:23)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:3284:73
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:10052:106)
    at StructurePanel (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621448429258:395088:121)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:3284:73
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:10052:106)
    at TransformerRegistry (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621448429258:395828:23)
    at ModuleRegistry (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621448429258:389542:23)
    at StructuredReportContent
    at TemplatePage
    at WithSuperJSON (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621448429258:326287:42)
    at BlitzInnerRoot (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:1386:5)
    at main
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:3284:73
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:10052:106)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:3284:73
    at Container (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:15730:86)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:3284:73
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:10052:106)
    at Layout (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621448429258:397513:20)
    at Suspense
    at ErrorBoundary (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:122016:37)
    at InnerThemeProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:50743:74)
    at ThemeProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:54908:5)
    at ThemeProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:50763:5)
    at SwitchableThemeProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:145283:23)
    at LocalizationProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:54737:5)
    at LocalizationProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:145042:23)
    at App (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:147630:24)
    at AppWithStructuredReportTranslations (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:146016:28)
    at AppWithSiteTranslations (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:145867:28)
    at Provider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:128488:20)
    at AppWithRedux (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:145732:32)
    at Hydrate (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:127760:23)
    at QueryClientProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:127831:21)
    at BlitzProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:1165:21)
    at BlitzOuterRoot (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621448429258:1417:34)
    at ErrorBoundary (http://localhost:3000/_next/static/chunks/main.js?ts=1621448429258:146:47)
    at ReactDevOverlay (http://localhost:3000/_next/static/chunks/main.js?ts=1621448429258:250:23)
    at Container (http://localhost:3000/_next/static/chunks/main.js?ts=1621448429258:8652:5)
    at AppContainer (http://localhost:3000/_next/static/chunks/main.js?ts=1621448429258:9141:24)
    at Root (http://localhost:3000/_next/static/chunks/main.js?ts=1621448429258:9277:25)
```

**Comment by DiegoMcDipster** [comment]
If you don't mind, i'll ask you a noob question, why is the second warning less of an issue than the first?

**Comment by medihack** [comment]
@DiegoMcDipster Both are issues during server-side rendering because of a mismatch between the rendered HTML on the server and the client. None of them is less or more important. The above pull request does fix both issues.

**Comment by DiegoMcDipster** [comment]
Ah.  OK.  Sorry, i'd misunderstood.  

Thanks @medihack. I'm new to opensource and github.  I'm really impressed how people help each other out to make the libraries better.

**Comment by medihack** [comment]
@DiegoMcDipster As long as the pull request is "Open" it is not merged. And then you still need to wait for a next release if you are installing it from NPM (but you can install it from a Github branch, too). Hope that helps and welcome to the community!

**Comment by DiegoMcDipster** [comment]
@medihack . Thanks for the welcome and also for the work you put into fixing this!

I thought that this was released and so i created a next app, installed the libraries via npm and tried the sample code on (https://docs.dndkit.com/presets/sortable#connecting-all-the-pieces). However, i'm still getting the error. Does this mean that it's not yet been released?
if that's the case, you said i can install via the Github branch.  Could you do me a massive favour and put the command that I need to run to install the branch please?

**Comment by medihack** [comment]
It is fixed and released with v3.0.4. Make sure you have the right version in package.json. And don't forget to give DndContext an id (required to fix this issue with ssr).

**Comment by DiegoMcDipster** [comment]
Fantastic!  It was the id that I was missing.  It's working a treat now!  Thanks again!
